lib/actions.js

```diff
@@ -637,12 +637,10 @@ exports.upload = function(selector, path) {
 	var self = this;
 	if (fs.existsSync(path)) {
 		return new HorsemanPromise(function(resolve, reject) {
-			return self.page.evaluate(function(selector) {
-				self.page.uploadFile(selector, path, impatient(function() {
-					resolve();
-					debug(".upload() " + path + " into " + selector);
-				}, self.options.timeout));
-			});
+            return self.page.uploadFile(selector, path, function() {
+                resolve();
+                debug(".upload() " + path + " into " + selector);
+            });
 		});
```

Comment:
Though, file system might be idling and the internal code couldn't check the given path for existence. Rare but possible.